### PR TITLE
bat: 0.11.0 -> 0.12.0

### PR DESCRIPTION
**Pull request by dywedir**
###### Motivation for this change
https://github.com/sharkdp/bat/releases/tag/v0.12.0

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @lilyball

### CODE REVIEW

**Comment by Ma27** [comment]
Built and tested locally :+1:

**Comment by Ma27** [comment]
@dywedir thanks!

File: pkgs/tools/misc/bat/default.nix [hunk 1]
```diff
@@ -1,25 +1,27 @@
...
-    sha256 = "0yyvlplskjvxb2cspqsvfsnahd5m0s83psrp777ng0wc0kr1adbw";
+    sha256 = "07qxghplqq8km4kp9zas2acw302a77y72x3ix1272kb1zxhw4as6";
     fetchSubmodules = true;
   };
 
-  cargoSha256 = "078n31c0isvxvna0s1m12xv4bkh15rb2nixfyg4c501mlkalb517";
+  cargoSha256 = "0j9wxv21a91yfvbbvgn5ms5zi1aipj1k2g42mfdvvw2vsdzqagxz";
 
-  nativeBuildInputs = [ cmake pkgconfig zlib ];
+  nativeBuildInputs = [ pkgconfig llvmPackages.libclang zlib ];
```
Comment (lilyball) [inline pkgs/tools/misc/bat/default.nix]: What's the reason for this change? The README still says you need `cmake`.

Comment (lilyball) [reply pkgs/tools/misc/bat/default.nix]: It does build on my system with this change. I'm not sure why `cmake` was required to begin with though, was it one of the `bat` dependencies? If it's not actually required then we should file an upstream issue to update the README.
